py/load_firmware.py

```diff
@@ -48,43 +66,58 @@ def main() -> int:
         eprint("Expecting signed firmware")
         return 1
 
-    bootloaders = devices.get_bitbox02_devices(devices.BB02_BOOTLOADER)
-    bootloaders.extend(devices.get_bitbox02_devices(devices.BB02BTC_BOOTLOADER))
-    bitboxes = devices.get_bitbox02_devices()
-
-    def _wait_for_bootloaders() -> List[devices.DeviceInfo]:
-        while True:
-            bootloaders = devices.get_bitbox02_devices(devices.BB02_BOOTLOADER)
-            bootloaders.extend(devices.get_bitbox02_devices(devices.BB02BTC_BOOTLOADER))
-            if bootloaders:
-                return bootloaders
-            sys.stdout.write(".")
-            sleep(1)
-
-    if not bootloaders:
-        if len(bitboxes) != 1:
-            eprint(
-                "No bitbox02 bootloader detected. Insert exactly one bootloader or "
-                "bitbox02 device."
-            )
-            return 1
+    bootloader = None
+    try:
+        bootloader = devices.get_bitbox02_bootloader()
+    except TooManyFoundException:
+        eprint("Found multiple bb02 bootloader standard editions. Only one supported.")
+
+    if bootloader is None:
+        try:
+            bootloader = devices.get_bitbox02btc_bootloader()
+        except TooManyFoundException:
+            eprint("Found multiple bb02 bootloader btc editions. Only one supported.")
+
+    if bootloader is None:
+        bitbox = None
+        try:
```

Comment:
a bit strange semantics, it will work if one device is plugged in, but also if 2 + 1 are plugged in.

What about having a function which returns all devices of all editions? The distinction is usually not important, and if it is we can have a `@property edition` to query for it per device/bootloader.

Reply:
ack